src/parsers/ddi_parser.jl

```diff
@@ -0,0 +1,174 @@
+
+
+"""
+    parse_ddi(filepath::String)
+
+Parses a valid IPUMs DDI XML file and returns a DDIInfo object containing 
+the IPUMs extract metadata. 
+
+### Arguments
+
+- `filepath::String` -- A string containing the path to the IPUMS DDI XML file. 
+
+### Returns
+
+A DDIInfo object that contains all of the file-level and variable-level 
+metadata for the IPUMs extract.
+
+# Examples
+
+Let's assume we have an extract DDI file named `my_extract.xml`
+```julia-repl
+julia> typeof(parse_ddi("my_extract.xml"))
+IPUMS.DDIInfo
+```
+"""
+function parse_ddi(filepath::String)
+
+    # check to make sure the provided file is an xml file.
+    splitext(filepath)[2] != ".xml" && throw(ArgumentError("The DDI file: $filepath should be an XML file."))
+    
+    # check to make sure file exists
+    !isfile(filepath) && throw(ArgumentError("The specified file: $filepath does not exist."))
+    
+    # QUESTION: check if file has multiple files -- then have to handle that
```

Comment:
Question: Since we concluded that it will probably be useful to have a bunch of these small "check" functions for datasets from IPUMS, do you want to replace these back with the calls to the appropriate small checker functions?

Reply:
Okay, I re-added the small checker functions. I agree, adding these small checker functions is probably easier to read compared to one-liners. Also these small checker functions help to document potential problems or issues we find in the data going forward.